src/api.ts

```diff
@@ -22,73 +22,52 @@ export async function explore(
   bundlesAndFileTokens: BundlesAndFileTokens,
   options: ExploreOptions = {}
 ): Promise<ExploreResult> {
-  const bundles: Bundle[] = [];
-  const fileTokens: string[] = [];
-
   bundlesAndFileTokens = Array.isArray(bundlesAndFileTokens)
     ? bundlesAndFileTokens
     : [bundlesAndFileTokens];
 
   if (bundlesAndFileTokens.length === 0) {
-    throw new AppError(ErrorCode.NoBundles);
+    throw new AppError({ code: 'NoBundles' });
   }
 
   // Separate bundles from file tokens
-  bundlesAndFileTokens.forEach(item => {
-    if (typeof item === 'string') {
-      fileTokens.push(item);
-    } else {
-      bundles.push(item);
-    }
-  });
+  const [fileTokens, bundles] = partition(
+    bundlesAndFileTokens,
+    (item): item is string => typeof item === 'string'
```

Comment:
The type guard isn't necessary here, `item => typeof item === 'string'` works fine.

Reply:
Removed type guard